crates/bitwarden/src/mobile/kdf.rs

```diff
@@ -10,10 +11,7 @@ pub async fn hash_password(
     password: String,
     kdf_params: Kdf,
 ) -> Result<String> {
-    let auth_settings = AuthSettings {
-        email,
-        kdf: kdf_params,
-    };
-    let hash = auth_settings.derive_user_password_hash(&password)?;
-    Ok(hash)
+    let master_key = MasterKey::derive(password.as_bytes(), email.as_bytes(), &kdf_params)?;
```

Comment:
We have this code repeated in multiple places now:
 - Here
 - In `make_register_keys`
 - In `determine_password_hash`
 - In `get_secret_verification_request`

Should we think of a way to avoid that repetition?

Reply:
Maybe. I think it's a good indication that we should try and pass along master key where it makes sense. I think we're fine with the duplication for now.